### PR TITLE
timeseries: persist and read settings

### PR DESCRIPTION
**Pull request by stephanwlee**
This is a continuation of #4991 where we are attempting to persist
several settings from TimeSeries to improve frustrating user experience
of having to reset settings over and over.

This change integrates the DataSource with the MetricsEffects so persist
new settings when it changes and bootstrap the application state from
the state in the LocalStorage.

Product choice: this change persist the smoothing value user may provide
in the URL since it felt more natural that way.

### CODE REVIEW

**Comment by stephanwlee** [comment]
To review: there was a large amount of white space change (indentation) so I would recommend turning on "Hide whitespace changes" option.

image

File: tensorboard/webapp/metrics/effects/metrics_effects_test.ts [hunk 1]
```diff
@@ -21,7 +21,7 @@ import {getActivePlugin} from '../../core/store';
 import * as coreTesting from '../../core/testing';
 import {DataLoadState} from '../../types/data';
 import {TBHttpClientTestingModule} from '../../webapp_data_source/tb_http_client_testing';
-import {of, Subject} from 'rxjs';
+import {EMPTY, of, Subject} from 'rxjs';
```
Comment (bmd3k) [inline tensorboard/webapp/metrics/effects/metrics_effects_test.ts:24]: There appears to be some significant refactoring in this CL mixed in with the new tests. Can we split the preliminary refactoring to a different PR?

Comment (stephanwlee) [reply tensorboard/webapp/metrics/effects/metrics_effects_test.ts:24]: Would splitting them into two separate commits help you review the PR? Also, did you turn on the "Hide whitespace changes" option on the GitHub? It is not a "refactor" and more of an indentation change :\

Comment (bmd3k) [reply tensorboard/webapp/metrics/effects/metrics_effects_test.ts:24]: Thanks for the tip. Indeed it was just whitespace. I totally missed your earlier comment about it, too.

**Review by bmd3k — APPROVED** [review]
